### PR TITLE
Add Edit Manga Details Modal

### LINKED ISSUE #76: Add Edit Manga Details Functionality

### PR DESCRIPTION
**Pull request by Animeboynz**
To Close #76

Chores:
* [x] Close Edit Modal After Save
* [x] Refresh Library after save (only if categories / custom fields have been edited)
* [x] Reset Entries when provided user input is blank

GUI:
* [x] Dates
* [x] Categories
* [x] Fork-Specifics
  
JS (if it actually changes the backup):
  * [x] Dates
  * [x] Categories
  * [x] Fork Specifics
    * [x] Title
    * [x] Author
    * [x] Artist
    * [x] Description
    * [x] Thumbnail
    * [x] Genre

### CODE REVIEW

**Comment by Animeboynz** [comment]
Just a very very basic proof of concept.

**Comment by BrutuZ** [comment]
* [ ] Make `#date-added` the right format too
* [ ] Make Categories a multi-select with names
Stretch Goals:
  * [ ] Introduce fork's custom fields: `customStatus`, `customThumbnailUrl`, `customTitle`, `customArtist`, `customAuthor`, `customDescription`, `customGenre`
    * [ ] Disable them if selected fork was Mihon?

**Comment by Animeboynz** [comment]
> [x] Make `#date-added` the right format too

It's done but with a bit of data loss. 
The timestamps for date added are 13 digit UNIX timestamps vs the 10 for the other details.
The last 3 digits are for milliseconds which gets dropped. Which I don't think is a big issue as anyone that is changing the date probably does not care about millisecond precision.

File: site/scripts/library.js [hunk 1]
```diff
@@ -72,10 +72,13 @@ export function initializeLibrary() {
 
       /////////////////////////////
       //const editCategoryOptions = document.getElementById("edit-category-options");
-      const option = document.createElement("option");
-      option.value = category.order;
-      option.textContent = category.name;
-      editCategoryOptions.appendChild(option);
+      if(category.order !== -1 && category.order !== 65535)
```
Comment (BrutuZ) [inline site/scripts/library.js:75]: Slightly less duplication
```suggestion
      if(![-1, 65535].includes(category.order))
```

**Comment by Animeboynz** [comment]
Currently when reading the timestamp it reads in GMT+0000 but when writing it uses the users timezone

File: site/scripts/library.js [hunk 2]
```diff
@@ -21,6 +21,11 @@ export function initializeLibrary() {
   let mangaItems = window.data.backupManga;
   const editCategoryOptions = document.getElementById("edit-category-options");
 
+  if (consts.fork.value === "sy")
```
Comment (BrutuZ) [inline site/scripts/library.js:24]:
```suggestion
  if (consts.fork.value !== 'mihon')
```

Not a SY exclusive

Comment (Animeboynz) [reply site/scripts/library.js:24]: I see, don't use many forks so no clue. Do they handle the additional info in the same way?

**Comment by Animeboynz** [comment]
Don't like the label alignment either. I was thinking it should be inline like in the settings but that would conflict with the description text area which we want as wide as possible. Maybe we could do the labels like this?
image

Also for genres I might do something similar to 
image

**Comment by BrutuZ** [comment]
I thought of just aligning the labels to the left/start, but these look nice, if implementing them doesn't add a lot more code or imports I'm cool with it

I have mixed feelings about that implementation for tags. It's close to what SY does, and while it looks nice I've wished a few times that it was a simple text field because it's annoying to remove and add one by one

**Comment by Animeboynz** [comment]
I don't really know what to do with CSS to make it look nice. Hopefully one of you can make my shitty attempt look better.
I'll finish up the JS today (next 4-6 hours) and hopefully we can just merge it soon.

**Pull request merged**